polaris-react/src/components/UnstyledLink/UnstyledLink.tsx

```diff
@@ -15,10 +15,10 @@ export interface UnstyledLinkProps extends LinkLikeComponentProps {}
 // but eslint-plugin-react doesn't know that just yet
 // eslint-disable-next-line react/display-name
 export const UnstyledLink = memo(
-  forwardRef<unknown, UnstyledLinkProps>(function UnstyledLink(props, _ref) {
+  forwardRef<any, UnstyledLinkProps>(function UnstyledLink(props, _ref) {
```

Comment:
I think we do similar elsewhere but avoiding `any` 

https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Banner/Banner.tsx#L272

https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Banner/Banner.tsx#L58

Reply:
Here are the combinations I've tried, and the resultant errors:

```tsx
forwardRef<unknown, UnstyledLinkProps>
```

<img width="1068" alt="Screenshot 2023-06-09 at 1 16 59 pm" src="https://github.com/Shopify/polaris/assets/612020/771d5474-4075-440e-8744-a319e0d52d24">

---

```tsx
forwardRef<HTMLAnchorElement, UnstyledLinkProps>
```
<img width="1489" alt="Screenshot 2023-06-09 at 1 18 30 pm" src="https://github.com/Shopify/polaris/assets/612020/3d69ac47-a5cb-4a39-9316-6c2bd38f5eb1">

---

```tsx
forwardRef<LinkLikeComponent, UnstyledLinkProps>
```

_(from https://github.com/Shopify/polaris/blob/main/polaris-react/src/utilities/link/types.ts#L18)_

<img width="1487" alt="Screenshot 2023-06-09 at 1 21 18 pm" src="https://github.com/Shopify/polaris/assets/612020/78293665-1eb1-482a-81f8-26d7b12281a8">

---

At this point I'm fresh out of ideas and just want to get this shipped so I can fix the actual bug and not TypeScript's pedanticism.

Reply:
@BPScott would this change affect web at all? I'm not sure why this was set to unknown in the first place but assuming it's web. Maybe we need a casting here `unknown as any` or some stronger typing if it will play nice in web